Improve floating-point number handling

tylerberry opened the pull request:
This pull request enables warnings for suspicious implicit casts, and fixes all warnings generated by enabling those warnings.

This involved switching from the use of float or double-specific functions (for example, floor vs floorf) to using CGFloat as the floating-point type and <tgmath.h> for the function prototypes. This results in code that works correctly and with the native floating-point width on both 32-bit and 64-bit platforms, and is warning-free!

tylerberry committed:
Enabled warnings for suspicious implicit casts, and fixed all warnings produced by this. This involved switching specific use of float or double functions to use CGFloat and <tgmath.h>.

DrummerB commented on BFColorPickerPopover/Extensions/NSColor+BFColorPickerPopover.m:
```diff
@@ -44,7 +44,7 @@ - (CGColorRef)copyCGColor
 
 + (NSColor *)randomColor {
-	return [NSColor colorWithCalibratedRed:(float)random()/RAND_MAX green:(float) random()/RAND_MAX blue:(float) random()/RAND_MAX alpha:1.0f];
+	return [NSColor colorWithCalibratedRed:random()/RAND_MAX green:random()/RAND_MAX blue:random()/RAND_MAX alpha:1.0];
```
Are you sure this is ok? random() and RAND_MAX are both integers, but we want floating point division. I can't test this right now.

tylerberry replied:
Good catch! I'll make a fixed pull request shortly.

tylerberry committed:
Cast to CGFloat before division to force floating point division. Also use arc4random instead of random, which I've seen Xcode warn about in other projects.

tylerberry commented:
Fixed to use an explicit cast to force FP division, which you correctly pointed out was necessary.

I also changed it to use arc4random() instead of random(), which is a "more random" random version. It's documented to return uint32_t, so UINT32_MAX is the correct divisor.

DrummerB commented:
Alright, merged it. Thanks for the contribution, Tyler!
